x-pack/plugins/fleet/server/tasks/unenroll_inactive_agents_task.ts

```diff
@@ -112,47 +113,54 @@ export class UnenrollInactiveAgentsTask {
       `[UnenrollInactiveAgentsTask] Fetching agent policies with unenroll_timeout > 0`
     );
     // find all agent policies that are not managed and having unenroll_timeout > 0
+    // limit the search to POLICIES_BATCHSIZE at a time and loop until there are no agent policies left
     const policiesKuery = `${AGENT_POLICY_SAVED_OBJECT_TYPE}.is_managed: false AND ${AGENT_POLICY_SAVED_OBJECT_TYPE}.unenroll_timeout > 0`;
-    const agentPolicies = await agentPolicyService.list(soClient, {
-      kuery: policiesKuery,
-      perPage: SO_SEARCH_LIMIT,
-    });
-    this.logger.debug(
-      `[UnenrollInactiveAgentsTask] Found "${agentPolicies.items.length}" agent policies with unenroll_timeout > 0`
-    );
-    if (!agentPolicies?.items.length) {
-      this.endRun('Found no policies to process');
-      return;
-    }
+    let hasMore = true;
+    let page = 1;
+    while (hasMore) {
+      const agentPolicies = await agentPolicyService.list(soClient, {
```

Comment:
Should it be better to use a point in time finder for that with the `fetchAllAgentPolicies`? it seems to me it will be more reliable

Reply:
Looking at [fetchAllAgentPolicies function](https://github.com/elastic/kibana/blob/99cb9b75f4f102fc08049bc88c39ec0ee71b6c4f/x-pack/plugins/fleet/server/services/agent_policy.ts#L1536) it seems that it fetches all the policies by kuery but without pagination? We had a previous [discussion](https://github.com/elastic/kibana/pull/189861#discussion_r1705590153) with @juliaElastic to fetch policies in batches of 500 policies and then iterate on them, it doesn't seem possible with this method. Is there a specific reason to use it instead of `list`?

Reply:
@criamico `fetchAllAgentPolicies` should return an async iterable (and you can configure the batch size with `perPage`)for example it's used here https://github.com/elastic/kibana/blob/main/x-pack/plugins/fleet/server/services/agent_policy_watch.ts#L84 
Using the point in time finder is I think a little more reliable and I think it guarantee to get all policies

Reply:
I switched `agentPoliciesService.list` to `fetchAllAgentPolicies`. Could you take a look again?